### PR TITLE
EPMRPP-79188 || Implement popover component

### PR DESCRIPTION
**Pull request by Vadim73i**

### CODE REVIEW

**Review by tr1ble — APPROVED** [review]

File: app/src/componentLibrary/popover/README.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+## **Popover with optional title and content**
+
+### Props:
+
+- **children**: _node_, optional, default = null
+- **title**: _string_, optional, default = ''
+- **trianglePosition**: _object_, optional, default = { side: 'top', position: 'left' }
+- **onClose**: _func_, optional, default = () => {}
+- **parentRef**: _object_, optional, default = null
+- **dataAutomationId**: _string_, optional, default = ''
```
Comment (AmsterGet) [inline app/src/componentLibrary/popover/README.md:10]: Please update readme according to the new naming.

File: app/src/componentLibrary/popover/withPopover/withPopover.jsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  const [isOpened, setOpened] = useState(false);
+
+  return (
+    <>
+      <div ref={parentRef} className={cx('with-popover')}>
+        <WrappedComponent
+          className={cx('wrapped-component')}
+          onClick={() => {
+            setOpened(true);
+          }}
```
Comment (AmsterGet) [inline app/src/componentLibrary/popover/withPopover/withPopover.jsx:36]: Are you sure that `WrappedComponent` will receive such props? Why not to handle click on parent `div`?
Please also don't forget to provide `{...props}` from the last anonymous function to the `WrappedComponent` as it is wrapped by HOC.

File: app/src/componentLibrary/popover/withPopover/withPopover.jsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        <Popover onClose={() => setOpened(false)} parentRef={parentRef} {...props}>
+          {content}
+        </Popover>
+      )}
+    </>
+  );
+};
+
+withPopover.propTypes = {
+  WrappedComponent: PropTypes.element.isRequired,
```
Comment (AmsterGet) [inline app/src/componentLibrary/popover/withPopover/withPopover.jsx:49]: `withPopover` is not a component, it's plain function.

File: app/src/componentLibrary/popover/withPopover/withPopover.jsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+import React, { useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import { Popover } from '../popover';
+import styles from './withPopover.scss';
+
+const cx = classNames.bind(styles);
+
+export const withPopover = ({ content, ...props }) => (WrappedComponent) => () => {
```
Comment (AmsterGet) [inline app/src/componentLibrary/popover/withPopover/withPopover.jsx:25]: It's not a `props`. Please rename to f.e. `popoverConfig`.

**Review by AmsterGet — CHANGES_REQUESTED** [review]

File: app/src/componentLibrary/popover/withPopover/withPopover.jsx [hunk 5]
```diff
@@ -15,41 +15,34 @@
...
 import classNames from 'classnames/bind';
 import { Popover } from '../popover';
 import styles from './withPopover.scss';
 
 const cx = classNames.bind(styles);
 
-export const withPopover = ({ content, ...props }) => (WrappedComponent) => () => {
+export const withPopover = ({ content, ...popoverConfig }) => (WrappedComponent) => ({
+  ...props
+}) => {
```
Comment (AmsterGet) [inline app/src/componentLibrary/popover/withPopover/withPopover.jsx:26]: Why not just `(props) => ...`?

**Review by AmsterGet — APPROVED** [review]

**Pull request merged**